src/triage/component/catwalk/evaluation.py

```diff
@@ -296,36 +381,59 @@ def _evaluations_for_group(
             )
         return evaluations
 
-    def evaluate(self, predictions_proba, matrix_store, model_id):
+    def evaluate(
+        self,
+        predictions_proba,
+        matrix_store,
+        model_id,
+        subset=None,
+    ):
         """Evaluate a model based on predictions, and save the results
 
         Args:
             predictions_proba (numpy.array) List of prediction probabilities
             matrix_store (catwalk.storage.MatrixStore) a wrapper for the
                 prediction matrix and metadata
             model_id (int) The database identifier of the model
+            subset (dict) A dictionary containing a predictions query and a
+                name for the subset to evaluate on, if any
         """
-        labels = matrix_store.labels()
+        # If we are evaluating on a subset, we want to get just the labels and
+        # predictions for the included entity-date pairs and write to the
+        # Test- or TrainSubsetEvaluations table, rather than the overall
+        # Test- or TrainEvaluations table.
+        if subset is not None:
+            evaluation_table_obj = matrix_store.matrix_type.subset_evaluation_obj
```

Comment:
What is the reason that the subsets should be their own table? Can we not model this with a nullable column on the evaluations table? Just to keep the existing post-modeling queries returning the same results as they do now? My counterpoint to this would be: if you are including subset evaluations in your experiment, you probably aren't going to be taken aback by the need to filter them because you're probably going to be looking to evaluate them anyway. If you aren't including subset evaluations in your experiment, these queries will return the same thing they do now.

A counterpoint to this might be: But then we have to make audition aware of this when it queries! Well, sure, but part of the reason for this change would be that audition can be more useful when it takes these into account, so this is work we're going to do anyway.

I'm not saying this PR would include audition changes (it could, although not necessary), but I figure we would put them in very soon to take full advantage of subsets.

Anyway, all this is rambling around my confusion around what a 'subset' is.

- Is it a slice of the predictions table? Kind of, but we only query the predictions table to get the index out. The code would be cleaner if we actually didn't have to take in an existing `predictions_proba` and just grabbed a slice of the predictions table. Side note: the evaluation code would be way cleaner if it just did this in the first place, though the extra database I/O may not be the best.
- Is it a cohort? No, as we are enforcing that they define it via from the predictions table.
- Is it a repurposing of the concept of an evaluation to optionally apply to a subset of the data? This to me is closest to what's implemented, but while they live in separate tables this mental model still isn't complete.

Reply:
Sure, we could add a nullable column to the evaluations table (and, yeah, audition will be changed anyway). I guess I just hadn't thought of making the column nullable and didn't want to have a reserved name (like `all`).

Yeah, "subset" like "model group" is probably a bit of a squishy concept that will change across contexts. In one project, it's just testing the usefulness of an overly-broad model on more restrictive program enrollment criteria. In another, it's that an inspections program will have enough person hours to inspect 40 properties in one district, 25 in another, 100 in a third, and so on, and we want to see how the precisions will be _in each district_ and how they will vary across districts if we use a single model for the whole program. The unifying thing in the subsets is the application of a model built on a larger cohort to just a smaller section of that cohort and re-calculation of evaluation metrics within that smaller section.